Components/Queues/Drivers/PhpAmqpLibDriver.php

```diff
@@ -0,0 +1,311 @@
+    {
+        $this->channel->basic_nack($message->getMessageId());
+    }
+
+    /**
+     * Creates a message and prepare the content to publish.
+     *
+     * {@inheritdoc}
+     * @throws \Exception
+     */
+    public function send(QueueMessageInterface $message, $destination = null, array $arguments = []): bool
+    {
+        $this->declareChannel();
+        $this->declareQueue($message->getQueue(), AMQP_DURABLE, $arguments);
+        $messageBody = $this->getSerializer()->serialize($message, $this->format);
+        $messageHeaders = new AMQPTable($message->getHeaders());
+        $amqpMessage = new AMQPMessage($messageBody);
+        $amqpMessage->set('application_headers', $messageHeaders);
+        $this->channel->basic_publish($amqpMessage, self::EXCHANGE_NAME, $message->getQueue());
+
+        return true;
+    }
+
+    /**
+     * Declares the queue to drive the message.
+     *
+     * @param string $queueName The name of the queue
+     * @param int $durable
+     * @param array $arguments See AMQPQueue::setArguments()
+     *
+     * @return array|null
+     */
+    public function declareQueue(string $queueName, int $durable = AMQP_DURABLE, array $arguments = [])
+    {
+        return $this->channel->queue_declare($queueName, false, $durable, false, false, false, new AMQPTable([$arguments]));
+    }
+
+    /**
+     * Catch a channel inside the connection.
+     *
+     * @param int $prefetchSize
+     * @param int $prefetchCount
+     * @return AMQPChannel
+     * @throws \Exception
+     */
+    public function declareChannel(int $prefetchSize = self::PREFETCH_SIZE, int $prefetchCount = self::PREFETCH_COUNT): AMQPChannel
+    {
+        if (!$this->channel instanceof AMQPChannel) {
+            $this->channel = $this->stream->channel();
+            $this->channel->basic_qos($prefetchSize, $prefetchCount, null);
+        }
+
+        return $this->channel;
+    }
+
+    /**
+     * Returns the format used on serialize function.
+     *
+     * @return string
+     */
+    public function getFormat(): string
+    {
+        return $this->format;
+    }
+
+    /**
+     * Set the format used by serialize/deserialize functions
+     *
+     * @param string $format
+     */
+    public function setFormat(string $format = null)
+    {
+        $this->format = $format;
+    }
+
+    /**
+     * Shuffles the connection in case there is more than one available.
+     */
+    protected function shuffleConnections(bool $shuffle = true)
+    {
+        if ($shuffle) {
+            shuffle($this->connectionsData);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function waitNoBlock()
+    {
+        $this->channel->wait(null, true);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function wait()
+    {
+        $this->channel->wait();
+    }
+
+    /**
+     * Checks if any messages are being consumed
+     *
+     * @return bool
+     */
+    public function isConsuming(): bool
```

Comment:
I think it is not needed to have this function as public. For what I can see it is only used inside this class

Reply:
The method were used just in the test functions. I replaced this calls for the native one and removed this method.